### PR TITLE
Fix broken tests on Windows + debug + MS STL

### PR DESCRIPTION
**Pull request by SergeyKopienko**

### CODE REVIEW

File: test/support/test_config.h [hunk 1]
```diff
@@ -120,6 +120,54 @@
...
+#define _PSTL_TEST_COMPLEX_TANH_MINUS_INF_NAN_BROKEN _PSTL_TEST_COMPLEX_OP_BROKEN
+#define _PSTL_TEST_COMPLEX_TANH_INF_INF_BROKEN _PSTL_TEST_COMPLEX_OP_BROKEN
+#define _PSTL_TEST_COMPLEX_TANH_NAN_MINUS_ZERO_BROKEN _PSTL_TEST_COMPLEX_OP_BROKEN
+
+#define _PSTL_TEST_COMPLEX_SIN_BROKEN _PSTL_TEST_COMPLEX_OP_BROKEN
+#define _PSTL_TEST_COMPLEX_COS_BROKEN _PSTL_TEST_COMPLEX_OP_BROKEN
+#define _PSTL_TEST_COMPLEX_LOG10_BROKEN _PSTL_TEST_COMPLEX_OP_BROKEN
+
+#define _PSTL_TEST_COMPLEX_DIV_COMPLEX_BROKEN _PSTL_TEST_COMPLEX_OP_BROKEN
+#define _PSTL_TEST_COMPLEX_TIMES_COMPLEX_BROKEN _PSTL_TEST_COMPLEX_OP_BROKEN
```
Comment (akukanov) [inline test/support/test_config.h:190]: Defining a separate macro for each test/operation makes no sense to me. Why is just one generic macro not enough?

File: test/xpu_api/numerics/complex.number/complex.transcendentals/tanh.pass.cpp [hunk 2]
```diff
@@ -61,19 +61,25 @@ void test_edges()
...
+#endif
         }
         else if (std::isnan(testcases[i].real()) && testcases[i].imag() == 0)
         {
+#if !_PSTL_TEST_COMPLEX_TANH_NAN_MINUS_ZERO_BROKEN
             assert(std::isnan(r.real()));
             assert(r.imag() == 0);
             assert(std::signbit(r.imag()) == std::signbit(testcases[i].imag()));
+#endif
         }
```
Comment (akukanov) [inline test/xpu_api/numerics/complex.number/complex.transcendentals/tanh.pass.cpp:83]: I'd skip the whole block (lines 62-83) instead; similarly in other tests.

File: test/support/test_config.h [hunk 3]
```diff
@@ -120,6 +120,54 @@
 #define TEST_HAS_NO_INT128
 #define _PSTL_TEST_COMPLEX_NON_FLOAT_AVAILABLE (_MSVC_STL_VERSION < 143)
 
+#define _PSTL_TEST_COMPLEX_OP_BROKEN (_MSVC_STL_VERSION && _MSVC_STL_VERSION <= 143)
```
Comment (akukanov) [inline test/support/test_config.h:144]: is the reason why it does not work well understood?

File: test/xpu_api/numerics/complex.number/cases.h [hunk 4]
```diff
@@ -19,175 +19,175 @@
...
-    dpl::complex<double>(-0., INFINITY),
-    dpl::complex<double>(+0., INFINITY),
-    dpl::complex<double>(0.5, INFINITY),
-    dpl::complex<double>(1, INFINITY),
-    dpl::complex<double>(2, INFINITY),
-    dpl::complex<double>(INFINITY, INFINITY)
+    dpl::complex<double>( 1.e-6,  1.e-6),       // 0
+    dpl::complex<double>(-1.e-6,  1.e-6),       // 1
+    dpl::complex<double>(-1.e-6, -1.e-6),       // 2
+    dpl::complex<double>( 1.e-6, -1.e-6),       // 3
```
Comment (akukanov) [inline test/xpu_api/numerics/complex.number/cases.h:25]: Is this numbering important?
It is hard/inconvenient to maintain. What if some extra tests will be added in the middle?

File: test/support/test_config.h [hunk 5]
```diff
@@ -168,7 +216,7 @@
 #    define _PSTL_ICPX_TEST_RED_BY_SEG_BROKEN_64BIT_TYPES 1
 #endif
 
-// Group reduction produces wrong results with multiplication of 64-bit for certain driver versions
+// Group reduction produces wrong results with multiplication of 64-bit  for certain driver versions
```
Comment (akukanov) [inline test/support/test_config.h:220]: Looks like an unintended sudden change.

File: test/support/test_config.h (hunk 3 above)
Comment (SergeyKopienko) [reply test/support/test_config.h:144]: As far as I seen these errors has place due MS standard library implementation.

File: test/support/test_config.h (hunk 1 above)
Comment (SergeyKopienko) [reply test/support/test_config.h:190]: This is our common approach in tests to have some separate macro for each broken case.
But or course I am ready to use `_PSTL_TEST_COMPLEX_OP_BROKEN` instead of them.

File: test/xpu_api/numerics/complex.number/complex.transcendentals/tanh.pass.cpp (hunk 2 above)
Comment (SergeyKopienko) [reply test/xpu_api/numerics/complex.number/complex.transcendentals/tanh.pass.cpp]: Disagree.
For each corner case in this test some separate check exist.
If we rewrite this part of code like
```C++
#if !_PSTL_TEST_COMPLEX_TANH_BROKEN
        else if (std::isinf(testcases[i].real()) && std::isinf(testcases[i].imag()))
        {
            assert(r.real() == (testcases[i].real() > 0 ? 1 : -1));
            assert(r.imag() == 0);
        }
        else if (std::isinf(testcases[i].real()) && std::isnan(testcases[i].imag()))
        {
            assert(r.real() == (testcases[i].real() > 0 ? 1 : -1));
            assert(r.imag() == 0);
        }
        else if (std::isnan(testcases[i].real()) && testcases[i].imag() == 0)
        {
            assert(std::isnan(r.real()));
            assert(r.imag() == 0);
            assert(std::signbit(r.imag()) == std::signbit(testcases[i].imag()));
        }
#endif
```
then we start other checks which has place in the code below.

File: test/support/test_config.h [hunk 6]
```diff
@@ -140,6 +140,23 @@
 #define _PSTL_ICC_TEST_COMPLEX_POLAR_BROKEN_TEST_EDGES __INTEL_LLVM_COMPILER
 #define _PSTL_ICC_TEST_COMPLEX_ISINF_BROKEN (_MSVC_STL_VERSION && __INTEL_LLVM_COMPILER)
 #define _PSTL_ICC_TEST_COMPLEX_ISNAN_BROKEN (_MSVC_STL_VERSION && __INTEL_LLVM_COMPILER)
+
+#define _PSTL_TEST_COMPLEX_OP_BROKEN (_MSVC_STL_VERSION && _MSVC_STL_VERSION <= 143)
```
Comment (danhoeflinger) [inline test/support/test_config.h:144]: I think we should include DEBUG in this macro, as these failures do not effect release mode in my experience.

Comment (danhoeflinger) [reply test/support/test_config.h:144]: Discussed offline, assert wont be checked in release /facepalm.

File: test/support/test_config.h (hunk 5 above)
Comment (SergeyKopienko) [reply test/support/test_config.h]: Fixed

File: test/xpu_api/numerics/complex.number/cases.h (hunk 4 above)
Comment (SergeyKopienko) [reply test/xpu_api/numerics/complex.number/cases.h]: Changes has been removed from this file

**Review by danhoeflinger — APPROVED** [review]
LGTM (pending green CI)

File: test/xpu_api/numerics/complex.number/complex.transcendentals/tanh.pass.cpp (hunk 2 above)
Comment (SergeyKopienko) [reply test/xpu_api/numerics/complex.number/complex.transcendentals/tanh.pass.cpp]: Fixed (where it's possible)

File: test/support/test_config.h [hunk 7]
```diff
@@ -140,6 +140,23 @@
...
+#define _PSTL_TEST_COMPLEX_ATANH_BROKEN _PSTL_TEST_COMPLEX_OP_BROKEN
+#define _PSTL_TEST_COMPLEX_COS_BROKEN _PSTL_TEST_COMPLEX_OP_BROKEN
+#define _PSTL_TEST_COMPLEX_COSH_BROKEN _PSTL_TEST_COMPLEX_OP_BROKEN
+#define _PSTL_TEST_COMPLEX_LOG10_BROKEN _PSTL_TEST_COMPLEX_OP_BROKEN
+#define _PSTL_TEST_COMPLEX_SIN_BROKEN _PSTL_TEST_COMPLEX_OP_BROKEN
+#define _PSTL_TEST_COMPLEX_SINH_BROKEN _PSTL_TEST_COMPLEX_OP_BROKEN
+#define _PSTL_TEST_COMPLEX_TANH_BROKEN _PSTL_TEST_COMPLEX_OP_BROKEN
+
+#define _PSTL_TEST_COMPLEX_TIMES_COMPLEX_BROKEN _PSTL_TEST_COMPLEX_OP_BROKEN
+#define _PSTL_TEST_COMPLEX_DIV_COMPLEX_BROKEN _PSTL_TEST_COMPLEX_OP_BROKEN
```
Comment (akukanov) [inline test/support/test_config.h:158]: I still do not understand the reason for this bunch of macros that just repeat another one.

Comment (SergeyKopienko) [reply test/support/test_config.h:158]: ...because we already  have this approach for another tests:
```C++
#define _PSTL_ICC_TEST_COMPLEX_ASIN_MINUS_INF_NAN_BROKEN_SIGNBIT __INTEL_LLVM_COMPILER
#define _PSTL_ICC_TEST_COMPLEX_COSH_MINUS_INF_MINUS_ZERO_BROKEN_SIGNBIT __INTEL_LLVM_COMPILER
#define _PSTL_ICC_TEST_COMPLEX_COSH_MINUS_ZERO_MINUS_ZERO_BROKEN_SIGNBIT __INTEL_LLVM_COMPILER
#define _PSTL_ICC_TEST_COMPLEX_POW_COMPLEX_COMPLEX_PASS_BROKEN_TEST_EDGES __INTEL_LLVM_COMPILER
#define _PSTL_ICC_TEST_COMPLEX_POW_COMPLEX_SCALAR_PASS_BROKEN_TEST_EDGES __INTEL_LLVM_COMPILER
#define _PSTL_ICC_TEST_COMPLEX_POW_SCALAR_COMPLEX_PASS_BROKEN_TEST_EDGES __INTEL_LLVM_COMPILER
#define _PSTL_ICC_TEST_COMPLEX_NORM_MINUS_INF_NAN_BROKEN_TEST_EDGES __INTEL_LLVM_COMPILER
#define _PSTL_ICC_TEST_COMPLEX_POLAR_BROKEN_TEST_EDGES __INTEL_LLVM_COMPILER
```

Comment (akukanov) [reply test/support/test_config.h:158]: And I believe back then I also told that it makes no sense to create many macros with exact same definition.

Comment (SergeyKopienko) [reply test/support/test_config.h:158]: OK, fix prepared: https://github.com/oneapi-src/oneDPL/pull/1569
